### PR TITLE
Improvements to application layout panel responsiveness

### PR DESCRIPTION
**Pull request by bartaz**
## Done

Improves responsive styling of application layout aside panel, provides built in width options and better documentation:
- makes sure aside panel gets unpinned on small screens
- makes sure aside panel doesn't exceed small screen width
- provides 3 width options for panels (is-narrow, is-medium, is-wide)
- improves documentation about application layout panels and customizing them

Fixes #3283

## QA

- Run `./run` or [demo](https://vanilla-framework-3397.demos.haus/docs/examples/)
- Check updated app layout documentation: https://vanilla-framework-3397.demos.haus/docs/layouts/application
- Check application layout panels example https://vanilla-framework-3397.demos.haus/docs/examples/layouts/application/structure
  - make sure aside panel works well responsively (unpins on small screens, is never wider than the screen)
  - check different built-in widths of the panel
- Check other new application layout examples with different panel setups: https://vanilla-framework-3397.demos.haus/docs/examples/

### CODE REVIEW

**Comment by barrymcgee** [comment]
@bartaz Looks great! I think it would be nicer to see the aside panel enter/exit with more animation though?

**Comment by bartaz** [comment]
> @bartaz Looks great! I think it would be nicer to see the aside panel enter/exit with more animation though?

Yes, we can add the animations. I added a separate issue for it: #3409

File: templates/docs/layouts/application.md [hunk 1]
```diff
@@ -10,51 +10,62 @@ context:
...
 
-The root of application layout, usually a direct child of the document `body` is an element with `l-application` class. This element defines the layout grid, all areas of the application layout need to be direct children.
+The root of the application layout is usually a direct child of the document `body`. It is denoted with the class `l-application`. This element defines the layout grid. All areas of the application layout need to be direct children of this root element.
 
 #### Navigation
 
-To provide fully responsive behaviour for small and large screen navigation. The navigation of this layout consists of two areas.
+The navigation of this layout consists of two areas:
+
+An element with class `l-navigation-bar` which represents a horizontal navigation bar visible on small screens only. It toggles the navigation sidebar drawer. We recommend using the `<header>` tag for this element, as this ensures it acts as an <a href="https://www.w3.org/TR/wai-aria-practices/examples/landmarks/region.html">ARIA region landmark</a>. E.g.: `<header class="l-navigation">`.
```
Comment (bartaz) [inline templates/docs/layouts/application.md:29]: @lyubomir-popov This is not correct. The `<header>` should be used for `l-navigation`, not `l-navigation-bar`. Also, next paragraph says about `l-navigation__drawer` as a direct child - this is child of `l-navigation`, not `l-navigation-bar`.

File: templates/docs/layouts/application.md [hunk 2]
```diff
@@ -10,69 +10,82 @@ context:
 
 ### Structure
 
-The application layout is a full-screen grid (represented by a root element with `l-application` class) consisting of several areas (represented by elements with class names such as `l-navigation`, `l-main`, etc.). The application layout styling defines the positioning and size of all the areas on different screen sizes. An example of this would be, the navigation sidebar will be fully expanded on large screens, but collapsible on small screens.
+The application layout is a css grid based layout that occupies the entire window. Its root element has a class called `l-application`. The layout defines several areas represented by elements with class names such as `l-navigation`, `l-main`, etc.
```
Comment (bartaz) [inline templates/docs/layouts/application.md:13]:
```suggestion
The application layout is a CSS grid based layout that occupies the entire window. Its root element has a class called `l-application`. The layout defines several areas represented by elements with class names such as `l-navigation`, `l-main`, etc.
```

**Comment by long-chung** [comment]
Bug in Chrome (V.86.0.4240.198) where text moves when you hover over any of the apps, units or machines

File: templates/docs/examples/layouts/application/JAAS.html [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+
+  <header class="l-navigation is-collapsed">
+    <div class="l-navigation__drawer">
+      <div class="p-panel is-dark">
+        <div class="p-panel__header--sticky">
+          <div class="u-fixed-width u-sv-2">
+            <div class="p-inline-list--stretch">
+              <div class="p-inline-list__item">
+                <a class="p-panel__logo" href="#">
+                  <img src="https://assets.ubuntu.com/v1/0b2b7bce-Vanilla-logo_88x24px_white.svg" alt="Vanilla framework" height="24">
```
Comment (anthonydillon) [inline templates/docs/examples/layouts/application/JAAS.html:40]: As a JAAS example, it would be nice to have the JAAS logo.

**Review by anthonydillon — APPROVED** [review]
Great work team. This works well and I can't wait to start using it :+1:

File: templates/docs/examples/layouts/application/JAAS.html (hunk 3 above)
Comment (bartaz) [reply templates/docs/examples/layouts/application/JAAS.html:40]: We will be updating logo and side navigation separately, so we can use the JAAS logo then.

Comment (anthonydillon) [reply templates/docs/examples/layouts/application/JAAS.html:40]: Ok, I noticed some issues with the side nav which I assumed was a follow-up. Though as this was just a logo would be ok but either works.

**Comment by clagom** [comment]
Just want to drop a "great work!" here too :)

File: templates/docs/layouts/application.md [hunk 4]
```diff
@@ -84,8 +97,74 @@ Below you can see an example of application layout with some basic panels applie
...
+
+On screens wider than `$application-layout--breakpoint-side-nav-expanded` (which defaults to the value of `$breakpoint-large`: `1036px`) the side navigation panel is expanded and always visible. Adding `is-pinned` class to `l-navigation` has no effect in this state (as the side navigation by default behaves as it was pinned).
+
+### Aside panel width
+
+Ideally, decisions about panel widths should be made once the content is known. That said, it is useful to have a few default values for prototyping purposes. The application layout provides 3 panel widths. These panel widths are derived from the static widths of 4, 8, 12 column containers as measured at window width bigger than `$grid-max-width`:
+
+- narrow: Matches the width of an 4 column container. Invoked with class `is-narrow`.
```
Comment (lyubomir-popov) [inline templates/docs/layouts/application.md:120]:
```suggestion
- narrow: Matches the width of a 4 column container. Invoked with class `is-narrow`.
```

File: templates/docs/layouts/application.md [hunk 5]
```diff
@@ -84,8 +97,74 @@ Below you can see an example of application layout with some basic panels applie
...
+- default: Matches the width of an 8 column container. No need for additional classed.
+- wide: Matches the width of 12 column container. Invoked with class `is-wide`.
```
Comment (lyubomir-popov) [inline templates/docs/layouts/application.md:122]:
```suggestion
- wide: Matches the width of a 12 column container. Invoked with class `is-wide`.
```

**Review by sowasred2012 — COMMENTED** [review]
Looks good to me - though there's one comment outstanding from @bartaz

**Review by sowasred2012 — APPROVED** [review]
LGTM

**Pull request merged**